### PR TITLE
Dockerizing component and deploying on aws 

### PR DESCRIPTION
**Pull request by Smarqu14**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,24 @@
+# What image do you want to start building on?
+FROM node:alpine
```
Comment (alexzhaohong) [inline Dockerfile:2]: Which version of alpine node?

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+# Tell your container where your app's source code will live
+WORKDIR  /src/GalleryComponent
+
+# What source code do you what to copy, and where to put it?
+COPY . /src/GalleryComponent
+
+# Does your app have any dependencies that should be installed?
+RUN npm install
+RUN apk update
```
Comment (alexzhaohong) [inline Dockerfile:15]: What is APK?

File: db/index.js [hunk 3]
```diff
@@ -1,8 +1,9 @@
 const mysql = require('mysql');
+
 const mysqlConfig = {
-  host: process.env.HOST,
-  user: process.env.USER,
-  password: process.env.PASSWORD,
+  host: 'db',
```
Comment (alexzhaohong) [inline db/index.js:4]: Are you no longer using your env variables?

File: db/seeder/seed.js [hunk 4]
```diff
@@ -3,8 +3,10 @@ const { s3Images } = require('./imageData.js');
 
 for (let i = 1; i <= 100; i += 1) {
   db.query('INSERT INTO products (product_id) VALUES (?)', [i]);
+  console.log('product ID at ===========', i);
   db.query('INSERT INTO images (image_url) VALUES (?)', [`${s3Images[Math.floor(Math.random() * 20)]}`]);
```
Comment (alexzhaohong) [inline db/seeder/seed.js:8]: great for debugging

File: docker-compose.yml [hunk 5]
```diff
@@ -0,0 +1,21 @@
+version: '3'
+services:
+  db:
+    image: mysql:5.7.29
+    environment:
+      MYSQL_DATABASE: ${DATABASE}
+      MYSQL_ROOT_PASSWORD: ${PASSWORD}
```
Comment (alexzhaohong) [inline docker-compose.yml:7]: Where are these variables coming from?

File: package.json [hunk 6]
```diff
@@ -8,21 +8,24 @@
     "node": ">=6.13.0"
   },
   "scripts": {
-    "reset-db": "mysql -u root < schema.sql",
+    "reset-db": "mysql -h 'db' -u root < schema.sql",
```
Comment (alexzhaohong) [inline package.json:11]: nice use of a flexible IP from docker compose

File: package.json [hunk 7]
```diff
@@ -8,21 +8,24 @@
...
+    "ci-test": "jest ./__tests__ --testPathIgnorePatterns ./server --ci -i --detectOpenHandles --forceExit",
+    "server": "sleep 30 && concurrently \"npm run reset-db\" \"npm run seed-db\" \"npm run start\" \"npm run react-dev\""
   },
   "dependencies": {
     "@fortawesome/fontawesome-svg-core": "^1.2.28",
     "@fortawesome/free-solid-svg-icons": "^5.13.0",
     "@fortawesome/react-fontawesome": "^0.1.9",
     "axios": "^0.19.2",
     "body-parser": "^1.19.0",
+    "concurrently": "^5.2.0",
```
Comment (alexzhaohong) [inline package.json:28]: I've never heard of this middleware

File: Dockerfile (hunk 1 above)
Comment (alexzhaohong) [reply Dockerfile:2]: I use 12-alpine

File: package.json (hunk 6 above)
Comment (Smarqu14) [reply package.json:11]: thanks

**Pull request merged**